### PR TITLE
INT-5828 - Improve performance of application ingestion

### PR DESCRIPTION
**Pull request by austinkelleher**

### CODE REVIEW

File: src/steps/applications.ts [hunk 1]
```diff
@@ -39,64 +45,211 @@ export async function fetchApplications({
...
+    executionHandler: fetchApplications,
+  },
+  {
+    id: Steps.BUILD_GROUP_APPLICATION_RELATIONSHIPS,
+    name: 'Build group application relationships',
+    entities: [],
     relationships: [
-      Relationships.ACCOUNT_HAS_APPLICATION,
       Relationships.GROUP_ASSIGNED_APPLICATION,
+      Relationships.USER_GROUP_ASSIGNED_AWS_IAM_ROLE,
```
Comment (zemberdotnet) [inline src/steps/applications.ts:239]: This is a Mapped Relationship right? Want to move it into that category.

File: src/steps/applications.ts [hunk 2]
```diff
@@ -39,64 +45,211 @@ export async function fetchApplications({
...
+  );
+}
+
 export const applicationSteps: IntegrationStep<IntegrationConfig>[] = [
   {
     id: Steps.APPLICATIONS,
     name: 'Fetch Applications',
     entities: [Entities.APPLICATION],
+    relationships: [Relationships.ACCOUNT_HAS_APPLICATION],
+    dependsOn: [Steps.GROUPS],
```
Comment (zemberdotnet) [inline src/steps/applications.ts:230]: Is this dependsOn still true?

File: docs/jupiterone.md [hunk 3]
```diff
@@ -167,24 +167,22 @@ The following entities are created:
...
-| `okta_app_user_group`                  | **HAS**               | `okta_user`           |
-| `okta_user_group, okta_app_user_group` | **ASSIGNED**          | `okta_application`    |
-| `okta_user_group`                      | **HAS**               | `okta_user`           |
-| `okta_rule`                            | **MANAGES**           | `okta_user_group`     |
-| `okta_user`                            | **ASSIGNED**          | `okta_application`    |
-| `okta_user`                            | **ASSIGNED**          | `aws_iam_role`        |
-| `okta_user`                            | **ASSIGNED**          | `mfa_device`          |
-| `okta_user`                            | **ASSIGNED**          | `okta_role`           |
-| `okta_user`                            | **CREATED**           | `okta_application`    |
-| `okta_user_group`                      | **ASSIGNED**          | `aws_iam_role`        |
```
Comment (zemberdotnet) [inline docs/jupiterone.md:186]: This seems to have dropped? Is this an issue?

Comment (austinkelleher) [reply docs/jupiterone.md:186]: Hm ya I'm not sure why this got dropped...The step metadata includes all of the same entities and relationships 🤔

File: src/steps/applications.ts (hunk 2 above)
Comment (austinkelleher) [reply src/steps/applications.ts:230]: Good comms. I'll remove.

File: src/steps/applications.ts (hunk 1 above)
Comment (austinkelleher) [reply src/steps/applications.ts:239]: good comms. will fix

File: src/steps/applications.ts (hunk 2 above)
Comment (zemberdotnet) [reply src/steps/applications.ts:230]: I guess I didn't think it all the way through, it does need a dependency on Steps.ACCOUNT

**Review by zemberdotnet — APPROVED** [review]

**Review by zemberdotnet — APPROVED** [review]

**Pull request merged**